custom_components/evohome_cc/__init__.py

```diff
@@ -158,8 +195,108 @@ async def load_system_config(store) -> Optional[Dict]:
         broker.update, hass_config[DOMAIN][CONF_SCAN_INTERVAL]
     )
 
+    setup_service_functions(hass, broker)
+
     return True
 
+@callback
+def setup_service_functions(hass: HomeAssistantType, broker):
+    @verify_domain_control(hass, DOMAIN)
+    async def force_refresh(call) -> None:
+        """Obtain the latest state data via the vendor's RESTful API."""
+        await broker.update()
+
+    @verify_domain_control(hass, DOMAIN)
+    async def set_system_mode(call) -> None:
+        """Set the system mode."""
+        payload = {
+            CONF_UNIQUE_ID: broker.client.evo.unique_id,
+            "service": call.service,
+            "data": call.data,
+        }
+        async_dispatcher_send(hass, DOMAIN, payload)
+
+    @verify_domain_control(hass, DOMAIN)
+    async def set_zone_override(call) -> None:
+        """Set the zone override (setpoint)."""
+
+        entity_id = call.data[ATTR_ENTITY_ID]
+
+        registry = await hass.helpers.entity_registry.async_get_registry()
+        registry_entry = registry.async_get(entity_id)
+
+        if registry_entry is None or registry_entry.platform != DOMAIN:
+            raise ValueError(f"'{entity_id}' is not a known {DOMAIN} entity")
+
+        if registry_entry.domain != "climate":
+            raise ValueError(f"'{entity_id}' is not an {DOMAIN} controller/zone")
+
+        payload = {
+            CONF_UNIQUE_ID: registry_entry.unique_id,
+            "service": call.service,
+            "data": call.data,
+        }
+
+        async_dispatcher_send(hass, DOMAIN, payload)
+
+    hass.services.async_register(DOMAIN, SVC_REFRESH_SYSTEM, force_refresh)
+
+    system_mode_schemas = []
+
+    # Not all systems support "AutoWithReset": register this handler only if required
+    if [m for m in SYSTEM_MODE_LOOKUP if m == EVO_MODE_RESET]:
+        hass.services.async_register(DOMAIN, SVC_RESET_SYSTEM, set_system_mode)
+
+    # These modes are set for a number of hours (or indefinitely): use this schema
+    temp_modes = [m for m in SYSTEM_MODE_LOOKUP if m == EVO_MODE_ECO]
+    if temp_modes:  # any of: "AutoWithEco", permanent or for 0-24 hours
+        schema = vol.Schema(
+            {
+                vol.Required(ATTR_SYSTEM_MODE): vol.In(temp_modes),
+                vol.Optional(ATTR_DURATION_HOURS): vol.All(
+                    cv.time_period,
+                    vol.Range(min=timedelta(hours=0), max=timedelta(hours=24)),
+                ),
+            }
+        )
+        system_mode_schemas.append(schema)
+
+    # These modes are set for a number of days (or indefinitely): use this schema
+    temp_modes = [m for m in SYSTEM_MODE_LOOKUP if m == EVO_MODE_AWAY or m == EVO_MODE_CUSTOM or m == EVO_MODE_DAY_OFF]
+    if temp_modes:  # any of: EVO_MODE_AWAY, EVO_MODE_CUSTOM, EVO_MODE_DAY_OFF, permanent or for 1-99 days
```

Comment:
If using `TEMP_MODES = ` (i.e. it's a constant), then the `if TEMP_MODES:` isn't needed at all?